app/assets/javascripts/admin/addon/controllers/admin-user-index.js

```diff
@@ -626,5 +559,57 @@ export default Controller.extend(CanCheckEmails, {
         }
       });
     },
+
+    showDeletePostsConfirmation() {
+      showModal("admin-delete-posts-confirmation", {
+        admin: true,
+        model: this.model,
+      });
+    },
+
+    deleteAllPosts() {
+      let deletedPosts = 0;
+      let deletedPercentage = 0;
+      const user = this.model;
+
+      const performDelete = (progressModal) => {
+        this.model
+          .deleteAllPosts()
+          .then(({ posts_deleted }) => {
+            if (posts_deleted === 0) {
+              user.set("post_count", 0);
+              progressModal.send("closeModal");
+            } else {
+              deletedPosts += posts_deleted;
+              deletedPercentage = Math.floor(
+                (deletedPosts * 100) / user.get("post_count")
+              );
+              progressModal.setProperties({
+                deletedPercentage: deletedPercentage,
+              });
+              performDelete(progressModal);
+            }
+          })
+          .catch((e) => {
+            progressModal.send("closeModal");
+            let error;
+            AdminUser.find(user.get("id")).then((u) => user.setProperties(u));
+            if (e.jqXHR.responseJSON && e.jqXHR.responseJSON.errors) {
```

Comment:
Can we not use `extractError` on this with a `defaultMessage`?  From `lib/ajax-error`

Reply:
Sure, I've made this change. Thanks for the tip!